pybamm/models/full_battery_models/base_battery_model.py

```diff
@@ -140,18 +140,18 @@ def default_var_pts(self):
     @property
     def default_submesh_types(self):
         base_submeshes = {
-            "negative electrode": pybamm.Uniform1DSubMesh,
-            "separator": pybamm.Uniform1DSubMesh,
-            "positive electrode": pybamm.Uniform1DSubMesh,
-            "negative particle": pybamm.Uniform1DSubMesh,
-            "positive particle": pybamm.Uniform1DSubMesh,
+            "negative electrode": pybamm.MeshGenerator1D(),
```

Comment:
I think this bit makes it a little unclear as to what mesh is being used in each domain. The following might reverse what we are trying to do though.. Maybe we could leave the mesh types as `pybamm.UniformSubMesh` etc but in somewhere else (maybe in `pybamm.Mesh`) create a dict of MeshGenereators and then use that dict. e.g.
In model:
```python
submesh_types = {"negative electrode": pybamm.UniformMesh1D, etc} 
```
In Mesh: 
```python
for domain in submesh_types.keys():
    if isinstance(submesh_types[domain], Submesh1D):
       submesh_generator[domain] = MeshGenerator1D(submesh_types[domain]) etc
    elif isinstance(submesh_types[domain], Submesh2D):
       submesh_generator[domain] = MeshGenerator2D(submesh_types[domain]) etc
```

Reply:
Alternative would be to leave the argument to MeshGenerator undefined